### PR TITLE
Second homework

### PR DESCRIPTION
**Pull request by yunusovAkmal**
Hello !
I have done second homework.
I have some questions and i would be grateful if you could answer them more detailed.

First,
app.use(urlencoded({ extended: false }));
what is difference when extended: //  false or true

Second,
why we need  // cors()

third,
what is mongoose Query ?
in console it is like object.
i have read docs,
it is not object, but i can use it like object,
it is not Promise, but i can use .then,
it is not func, but it can be executed, or maybe i  misunderstood
can you explain ? // more detailed

fourth,
as you said i changed type for module in package.json and used ES modules, but encountered with some weird situations.
I had to rename all my routes, models, middlewares format from .js to .mjs
and import from
// import homeR from "./routes/home";
to 
// import homeR from "./routes/home.mjs";
or // import auth from "../middleware/verify.js";
as it gave error // code: 'ERR_MODULE_NOT_FOUND'
1Q. when type = module, is it mandatory to import like this?
2Q. what file extension is convenient // .js  or // .mjs  and when?

and also importing 
like this // import {Schema, model} from "mongoose";
doesn't work, so i changed it to this
// import mongoose from "mongoose";
// const { Schema, model } = mongoose; 
but this is working // import { Router } from "express";
3Q.  how this worked / not worked?

fifth,
what is more preferable // findOneAndUpdate  or  updateOne ?

Thanks !

### CODE REVIEW

File: middleware/passport.mjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+const opts = {
+	jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+	secretOrKey: process.env.JWT_SECRET,
+};
+
+export default function (passport) {
+	passport.use(
+		new Strategy(opts, async (payload, done) => {
+			const user = await User.findById(payload.id);
+			if (user) {
```
Comment (devbarden) [inline middleware/passport.mjs:13]: instead of this condition you can rewrite it in one line
`done(null, user || false)` smth like that

**Review by devbarden — COMMENTED** [review]
Not bad, but I advice you split code in each router. You should create controllers for each router. For example: 
`router.get("/", auth, AccountController.findAll)`
and then in controllers/AccountController you can create functions or Class for example

**Comment by devbarden** [comment]
Regarding your questions. Please use google) Most of your questions easy to find at stackoverflow, google, habr and etc. Good Programmer === Good Google Searcher :)
If you won't find then you can ask, ok ? Just try to google on English.
Questions 1,2,3 easy to google
4. it depends which node js version you are using. also you can search this problem at google and if you won't find the solution then you can go back to common.js modules but actually all of your code should work with latest node js and without .mjs file
5. just 'update' will be enough.. or 'updateById'

File: middleware/passport.mjs (hunk 1 above)
Comment (yunusovAkmal) [reply middleware/passport.mjs:13]: I fixed code and created controllers, can i do merge

Comment (devbarden) [reply middleware/passport.mjs:13]: sure

**Pull request merged**